benchexec/test_core_assignment.py

```diff
@@ -100,601 +186,319 @@ def test_singleThread(self):
-            )
+        self.t_unit_assertValid(3, self.threeCore_assignment)
 
     def test_fourCoresPerRun(self):
         # test all possible numOfThread values for runs with four cores
-        maxThreads = self.cpus * (self.cores // 4)
-        if not self.use_ht and self.ht:
-            maxThreads = self.cpus * (self.cores // 8)
-            if maxThreads == 0:
-                # Test for runs that are split over cpus
-                cpus_per_run = int(math.ceil(4 / (self.cores // 2)))
-                maxThreads = self.cpus // cpus_per_run
-
-        self.assertInvalid(4, maxThreads + 1)
-        if not self.fourCore_assignment:
-            self.skipTest("Need result specified")
-        for num_of_threads in range(1, maxThreads + 1):
-            self.assertValid(
-                4, num_of_threads, self.fourCore_assignment[:num_of_threads]
-            )
+        self.t_unit_assertValid(4, self.fourCore_assignment)
 
     def test_eightCoresPerRun(self):
         # test all possible numOfThread values for runs with eight cores
-        maxThreads = self.cpus * (self.cores // 8)
-        if not self.use_ht and self.ht:
-            maxThreads = (self.cpus * self.cores) // 16
-            if maxThreads == 0:
-                # Test for runs that are split over cpus
-                cpus_per_run = int(math.ceil(8 / (self.cores // 2)))
-                maxThreads = self.cpus // cpus_per_run
-        if not maxThreads:
-            self.skipTest(
-                "Testing for runs that need to be split across CPUs is not implemented"
-            )
-        self.assertInvalid(8, maxThreads + 1)
-        if not self.eightCore_assignment:
-            self.skipTest("Need result specified")
-        for num_of_threads in range(1, maxThreads + 1):
-            self.assertValid(
-                8, num_of_threads, self.eightCore_assignment[:num_of_threads]
-            )
+        self.t_unit_assertValid(8, self.eightCore_assignment)
 
 
-class TestCpuCoresPerRun_singleCPU(TestCpuCoresPerRun):
-    cpus = 1
-    cores = 8
-    ht = False
+class Test_Topology_P1_NUMA2_L8_C16_F(TestCpuCoresPerRun):
+    num_of_cores = 16
+    num_of_packages = 1
+    num_of_NUMAs = 2
+    num_of_L3_regions = 8
+    num_of_hyperthreading_siblings = 2
+    use_hyperthreading = False
 
-    oneCore_assignment = [[x] for x in range(8)]
-    twoCore_assignment = [[0, 1], [2, 3], [4, 5], [6, 7]]
-    threeCore_assignment = [[0, 1, 2], [3, 4, 5]]
-    fourCore_assignment = [[0, 1, 2, 3], [4, 5, 6, 7]]
-    eightCore_assignment = [list(range(8))]
+    """                 x
 
-    def test_singleCPU_invalid(self):
-        self.assertInvalid(2, 5)
-        self.assertInvalid(5, 2)
-        self.assertInvalid(3, 3)
+            x                       x
 
+    x   x       x   x       x   x      x    x
 
-class TestCpuCoresPerRun_singleCPU_HT(TestCpuCoresPerRun_singleCPU):
-    ht = True
+    x-  x-      x-  x-      x-  x-     x-  x-
```

Comment:
```
    x : symbolizes a unit (package, NUMA, L3, core)
    - : visualizes that a core is there, but it is not available because
        use_hyperthreading is set to False
```

Reply:
For better understanding, this is edited again in https://github.com/sosy-lab/benchexec/pull/892/commits/0689df517323f46784726a9fc5b3d74c0abcdad0 

    x : symbolizes a unit (package, NUMA, L3, core)
    - : visualizes that a core is there, but it is not available because
        use_hyperthreading is set to False
    int: core id

                         x

            x                       x

    x   x       x   x       x   x      x    x

    0-  2-     4-   6-     8-   10-   12-  14-